dotcom-rendering/src/lib/identity.ts

```diff
@@ -0,0 +1,47 @@
+import { IdentityAuth } from '@guardian/identity-auth';
+
+function getStage() {
+	if (!window.guardian.config.isDev) {
+		return window.guardian.config.stage;
+	} else return 'DEV';
+}
+
+const getIssuer = (stage: StageType) =>
+	stage === 'PROD'
+		? 'https://profile.theguardian.com/oauth2/aus3xgj525jYQRowl417'
+		: 'https://profile.code.dev-theguardian.com/oauth2/aus3v9gla95Toj0EE0x7';
+
+const getClientId = (stage: StageType) =>
+	stage === 'PROD' ? '0oa79m1fmgzrtaHc1417' : '0oa53x6k5wGYXOGzm0x7';
+
+const getRedirectUri = (stage: StageType) => {
+	switch (stage) {
+		case 'PROD':
+			return 'https://www.theguardian.com/ ';
+		case 'CODE':
+			return 'https://m.code.dev-theguardian.com/';
+		case 'DEV':
+		default:
+			return 'http://localhost:3030/';
+	}
+};
+
+let identityAuth: IdentityAuth | undefined = undefined;
+
+function getIdentityAuth() {
+	if (identityAuth === undefined) {
+		const stage = getStage();
+
+		identityAuth = new IdentityAuth({
+			issuer: getIssuer(stage),
+			clientId: getClientId(stage),
+			redirectUri: getRedirectUri(stage),
+			scopes: ['openid', 'profile', 'email'], // and any other scopes you need
+		});
+	}
+	return identityAuth;
+}
+
+export async function isSignedInWithOkta(): Promise<boolean> {
+	return getIdentityAuth().isSignedIn();
```

Comment:
this could then become 
```suggestion
	return identityAuth.isSignedIn();
```

Reply:
`isSignedIn` only returns a boolean though, which is fine for just checking the current signed in state.

It might be more flexible in the future to use the [`isSignedInWithAuthState`](https://github.com/guardian/csnx/tree/main/libs/%40guardian/identity-auth#issignedinwithauthstate) method which returns an [`IdentityAuthState`](https://github.com/guardian/csnx/blob/547b5f17a2094543fe82cc08b4d14610fae23db6/libs/%40guardian/identity-auth/src/%40types/OAuth.ts#LL26C4-L26C4) type, which includes the boolean (`isAuthenticated`), as well as the tokens, which might be required to read information about the user (using the id token), or making calls to external apis (using the access token).